### ISSUE TITLE
Reading 2D array branch modifies other return branch shape

### ISSUE THREAD

**Issue by agoose77** [scikit-hep/uproot4#244]
```python
>>> import uproot
>>> uproot.__version__
4.0.1
```

I process a lot of data sequentially (because the operations are not trivial); here is a simple function to pull out a list of batches from a test file (N.B. I would not use this in production!)

```python
import itertools
import pathlib
import uproot

path = pathlib.Path.cwd() / "tracks.txt"

def load_branches(path, branches):
    with uproot.open(f"{path}:tracks") as tracks_file:
        tracks_batches = tracks_file.iterate(
            branches,
            entry_start=0,
            entry_stop=100,
            step_size="50kB",
            library="ak"
        )
        return list(tracks_batches)
```
When I read the 2D covariance matrix branch from this test file, the shape of the returned `v_y` branch is modified:
```python
>>> batch, = load_branches(path, ['v_y'])
>>> batch['v_y'][0]
147.14722347885558
```

Introducing the covariance branch:
```python
>>> batch, = load_branches(path, ['v_y', 'covariance'])
>>> batch['v_y'][0]
<Array [... 126, 111, 54.3, 170, 169, 114]] type='1 * 100 * float64'
```

ROOT file:
[tracks.txt](https://github.com/scikit-hep/uproot4/files/5829933/tracks.txt)

**Comment by jpivarski** [comment]
This should be fixed in PR #245. The reason this happened is because [ak.zip](https://awkward-array.readthedocs.io/en/latest/_auto/ak.zip.html) had a surprising behavior if its arguments were regular arrays (like your `covariance`). They were "[right broadcasted](https://awkward-array.readthedocs.io/en/latest/_auto/ak.broadcast_arrays.html)," a feature intended for consistency with NumPy when computing ufuncs, but it doesn't really belong in `ak.zip` (though they both use the same function to do broadcasting—a consistency that is unwanted in this case!).

I ~~changed~~ fixed `ak.zip` in the Awkward Array library (scikit-hep/awkward-1.0#656) and also avoided using `ak.zip` in Uproot where an [ak.Array dict constructor](https://awkward-array.readthedocs.io/en/latest/_auto/ak.Array.html) would do the same thing. That way, you get the right behavior regardless of whether you've updated Uproot, Awkward Array, or both.

**Comment by agoose77** [comment]
Thanks for the lightning fast response on this Jim, it's much appreciated! :tada:
